imagetest/test_suites/image_validation/license_test.go

```diff
@@ -0,0 +1,190 @@
+	`Permission is hereby granted.*to any person obtaining a copy of.*and associated documentation files.*to deal in.*without restriction`,
+	`Redistribution and use of this software and associated documentation ("Software"), with or without modification, are permitted`,
+	`This code is multi Licensed under all/any one of.*LGPLv2.*New Style BSD.*MIT`,
+	`LICENSE. You may copy and use the Software, subject to these conditions: 1. This Software is licensed for use only in conjunction with Intel component products. Use of the Software in conjunction with non-Intel component products is not licensed hereunder.`,
+	`Brocade Linux Fibre Channel HBA Firmware`,
+	`QLogic Linux Fibre Channel HBA Firmware`,
+	`Unlimited distribution and/or modification is allowed as long as this copyrightPathGlob notice remains intact.`,
+	`Permission is hereby granted to use.*this.*for any purpose`,
+	`are in the public domain`,
+	`is (available|distributed) under the terms of the GNU.*Public License`,
+	`(libudev|libgudev|udev) is licensed under the GNU (L|)GPL`,
+	`The Linux Console Tools are covered by the GPL`,
+	`Some portions of os-prober`,
+	`Netcat and the associated package is a product of Avian Research, and is freely available in full source form with no restrictions save an obligation to give credit where due.`,
+	`Permission is hereby granted, without written agreement and without licence or royalty fees, to use, copy, modify, and distribute this software`,
+	`Open Market permits you to use, copy, modify, distribute, and license this Software and the Documentation for any purpose, provided that existing copyrightPathGlob notices are retained in all copies and that this notice is included verbatim in any distributions. No written agreement, license, or royalty fee is required for any of the authorized uses.`,
+	`This software is made available under the terms of *either* of the licenses found in LICENSE.APACHE or LICENSE.BSD. Contributions to cryptography are made under the terms of *both* these licenses.`,
+}
+
+const (
+	copyrightPathGlob = "/usr/share/doc/*/copyrightPathGlob"
+	licensePathGlob   = "/usr/share/doc/*/LICENSE"
+	licenseNameRegex  = `(?i)((?:(?:License|Copyright)\s*:\s*%[1]s)|(?:(?:covered )*under (?:the )?%[1]s)|(?:under (?:the terms of )*the %[1]s))`
+)
+
+func isValidLicenseName(licenseCheck string) bool {
+	for _, name := range licenseNames {
+		// (?i) case insensitive
+		var regexString = fmt.Sprintf(licenseNameRegex, name)
+		re := regexp.MustCompile(regexString)
+
+		if re.MatchString(licenseCheck) {
+			return true
+		}
+	}
+	return false
+}
+
+func isValidLicenseText(licenseCheck string) bool {
+	for _, licenseText := range licenses {
+		// (?i) case insensitive
+		re := regexp.MustCompile(`(?i)` + licenseText)
+
+		if re.MatchString(licenseCheck) {
+			return true
+		}
+	}
+	return false
+}
+
+func TestArePackagesLegal(t *testing.T) {
+	var filenames []string
+	if utils.IsTargetLinux(utils.RedHat) || utils.IsTargetLinux(utils.SUSE) {
```

Comment:
i usually think a sequences of ifs against the same values are best handled with a switch statement. maybe like:
```
image, err := utils.GetMetadata("image")
if err != nil {
  t.Fatalf("couldn't get image from metadata")
}
switch {
case strings.Contains(image, "rhel"), strings.Contains(image, "suse"):
  pathGlob = licensePathGlob
default:
  pathGlob = copyrightPathGlob
}
filenames, err := filepath.Glob(pathGlob)
if err != nil {
  t.Fatalf("couldnt resolve glob %s", pathGlob)
}
for _, filename := range filenames {
```